src/Microsoft.Azure.NotificationHubs/Messaging/MessagingUtilities.cs

```diff
@@ -0,0 +1,77 @@
+﻿using System;
+using System.Linq;
+using System.Collections.Generic;
+
+namespace Microsoft.Azure.NotificationHubs.Messaging
+{
+    public static class MessagingUtilities
+    {
+        public static void ThrowIfNullAddressOrPathExists(Uri address, string paramName)
```

Comment:
I believe we can simplify this:

 ```suggestion
        public static void ThrowIfNullAddressOrPathExists(Uri address)
        {
           ...
           throw new ArgumentNullException(nameof(address));
        }
```